alkira/resource_alkira_peering_gateway_cxp.go

```diff
@@ -0,0 +1,193 @@
+				Required: true,
+			},
+
+			// TODO: change cloud_provider to be a Required value when more cloud providers are added and remove the default value.
+			"cloud_provider": {
+				Description: "The cloud provider on which the resource will be created." +
+					"Default value is AZURE and only AZURE is supported for now.",
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "AZURE",
+			},
+			"cloud_region": {
+				Description: "The region of the specified cloud provider on which the resource should be created." +
+					"E.g. if cloud_provider is AZURE, the region could be eastus.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"segment_id": {
+				Description: "The ID of the segment in which the gateway is created." +
+					"Once deployed, this property can not be changed.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"state": {
+				Description: "The state of the resource.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceAlkiraPeeringGatewayCxpCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// INIT
+	api := alkira.NewPeeringGatewayCxp(m.(*alkira.AlkiraClient))
+
+	request, err := generateAlkiraCxpPeeringGatewayRequest(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// CREATE
+	response, _, err, _ := api.Create(request)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(string(response.Id))
+
+	// WAIT FOR STATE
+	state := response.State
+
+	for state != "ACTIVE" {
+		resource, _, err := api.GetById(d.Id())
+		if err != nil {
+			return diag.Diagnostics{{
+				Severity: diag.Warning,
+				Summary:  "FAILED TO GET RESOURCE",
+				Detail:   fmt.Sprintf("%s", err),
+			}}
+		}
+
+		state = resource.State
+		time.Sleep(5 * time.Second)
+	}
+
+	return resourceAlkiraPeeringGatewayCxpRead(ctx, d, m)
+}
+
+func resourceAlkiraPeeringGatewayCxpRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// INIT
+	api := alkira.NewPeeringGatewayCxp(m.(*alkira.AlkiraClient))
+
+	resource, _, err := api.GetById(d.Id())
+	if err != nil {
+		return diag.Diagnostics{{
+			Severity: diag.Warning,
+			Summary:  "FAILED TO GET RESOURCE",
+			Detail:   fmt.Sprintf("%s", err),
+		}}
+	}
+	segmentId, err := getSegmentIdByName(resource.Segment, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.Set("name", resource.Name)
+	d.Set("description", resource.Description)
+	d.Set("cxp", resource.Cxp)
+	d.Set("cloud_provider", resource.CloudProvider)
+	d.Set("cloud_region", resource.CloudRegion)
+	d.Set("segment_id", segmentId)
+	d.Set("state", resource.State)
+
+	return nil
+}
+
+func resourceAlkiraPeeringGatewayCxpUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// checking if only description field is changed
+	if d.HasChange("description") && !d.HasChanges("name", "cloud_provider", "cloud_region", "segment_id") {
```

Comment:
Honestly, if the only field could be changed is description, we should just fail and return. It's pointless to just check description. Just return directly. Simple and clean.